### ISSUE TITLE
Config not loading on .NET Core console application

### ISSUE THREAD

**Issue by Swoogan** [NLog/NLog#4144]
**NLog version**: 4.7.5

**Platform**: .NET Core 3.1

**Current NLog config** (xml or C#, if relevant)

```xml
<?xml version="1.0" encoding="utf-8" ?>
<nlog xmlns="http://www.nlog-project.org/schemas/NLog.xsd"
      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">

    <targets>
        <target name="logfile" xsi:type="File" fileName="file.txt" />
        <target name="logconsole" xsi:type="Console" />
    </targets>

    <rules>
        <logger name="*" minlevel="Info" writeTo="logconsole" />
        <logger name="*" minlevel="Debug" writeTo="logfile" />
    </rules>
</nlog>
```

- What is the current result?
No message appear in console. No log file is created

- What is the expected result?
Messages appear on the console and/or log file is created

- Did you checked the [Internal log](https://github.com/NLog/NLog/wiki/Internal-Logging)?
This file was not created either

- Please post full exception details (message, stacktrace, inner exceptions)
N/A

- Are there any workarounds?
Add `LogManager.LoadConfiguration("nlog.config");` to the program

- Is there a version in which it did work? N/A
- Can you help us by writing an unit test? N/A

I followed the process I am familiar with on .NET framework. I also read the .NET Core console tutorial, but I don't use DI or the Microsoft.Extensions.Logging framework so it wasn't entirely useful: https://github.com/NLog/NLog/wiki/Getting-started-with-.NET-Core-2---Console-application

It says if you don't use those tools, to follow the .NET Framework tutorial found here: https://github.com/NLog/NLog/wiki/Tutorial
This tutorial does not produce logging unless you use the workaround I posted. The .NET tutorial also does not list that you have to manually add the following to your csproj:

     <ItemGroup>
         <None Update="nlog.config" CopyToOutputDirectory="PreserveNewest" />
     </ItemGroup>

Anyway, this isn't so much a bug as the documentation bouncing you around and never giving the full picture. I think it would be best to have a .NET Core Console app tutorial that doesn't include DI and Microsoft.Extensions.Logging, then link the existing tutorial as an "advanced setup" or "logging with DI".

**Comment by snakefoot** [comment]
@Swoogan Can you include the output from the NLog InternalLogger:

```c#
using NLog;
 
class Program
{
  static void Main()
  {
    NLog.Common.InternalLogger.LogFile = "c:\\nlog-internal-log.txt";
    NLog.Common.InternalLogger.LogLevel = LogLevel.Debug;
    var logger = NLog.LogManager.GetCurrentClassLogger();
  }
}
```

And then attach the output to this issue.

**Comment by snakefoot** [comment]
@Swoogan So the trouble is actually the missing detail about using `CopyToOutputDirectory` in the [wiki](https://github.com/NLog/NLog/wiki/Getting-started-with-.NET-Core-2---Console-application)

**Comment by Swoogan** [comment]
Well, I thought the issue was that it doesn't work without `LogManager.LoadConfiguration("nlog.config");`

But I removed that and ran your code and there were no errors, so I am a little confused. I went back to an earlier revision of my code (where logging wasn't working) and ran it and found that my config wasn't valid. But what made it invalid was a change that I made because logging wasn't working.

So in summary, I don't know what I was doing wrong. What didn't work, now works. So any conclusions about the source of the problem are now invalid. It could be that I'm on Linux and something like "C:\\internal-log.txt" or `${baseDir}` aren't valid, but it's tough to say what I had wrong.

**Comment by snakefoot** [comment]
You can try this instead on Linux:

```c#
using NLog;
 
class Program
{
  static void Main()
  {
    NLog.Common.InternalLogger.LogFile = "/home/nlog-internal-log.txt";
    NLog.Common.InternalLogger.LogLevel = LogLevel.Debug;
    var logger = NLog.LogManager.GetCurrentClassLogger();
  }
}
```

**Comment by Swoogan** [comment]
I feel that with the edit made to the wiki tutorial this issue is now resolved. I have updated my SO answer to reflect the new information. Thank you for your help. The sample code with the `InternalLogger` configuration is a lifesaver. My `nlog-internal-log.txt` does not contain anything interesting because everything is now working as expected.

**Comment by snakefoot** [comment]
Happy to hear. Will close the issue as resolved then.
